src/clj_jargon/init.clj

```diff
@@ -16,9 +16,10 @@
 (def ^:dynamic curr-with-jargon-index nil)
 
 (defn clean-return
-  [{^IRODSFileSystem cm-proxy :proxy} retval]
+  [cm retval]
   (log/debug curr-with-jargon-index "- cleaning up and returning a plain value")
-  (.close cm-proxy)
+  (when (and (delay? cm) (realized? cm)) (.close (:proxy @cm)))
```

Comment:
i.e.: don't bother closing the cm if it was never actually opened! also, don't NPE when it's a delay, heh